### PR TITLE
docs(ai-history): draft ch29 support vector machines (#403)

### PR DESCRIPTION
**Pull request by krisztiankoos**
## Summary

- drafts Ch29 Support Vector Machines from the approved research contract
- lands at 4,015 words after prose-review tightening, using the Ch24/Ch28 prose shape and verified VC/SVM/OCR anchors
- updates Ch29 status to prose review with `prose_words: 4015`

## Review Follow-Up

- Claude approved and independently checked the OCR/source anchors.
- Gemini first pass requested removal of prompt/meta leakage and less rhetorical repetition.
- Follow-up commit `723645ad` removed the process note/meta-language, tightened the draft, and applied Claude's OCR nits.
- Gemini Flash follow-up found two remaining meta references; commit `ea80680b` removed them and changed NIST wording to "setup".
- Final Gemini follow-up approved `ea80680b` after gemini-3.1-pro-preview timed out and the bridge used the requested gemini-3-flash-preview fallback.

## Guardrails Carried Into Prose

- no "SVMs defeated neural networks" or "1990s winner" story
- VC theory is conceptual only; no theorem derivation beyond verified anchors
- LeNet comparison is attributed to Cortes/Vapnik's report
- no unsupported lab/hardware color or modern scikit-learn detour

## Checks

- `wc -w src/content/docs/ai-history/ch-29-support-vector-machines.md` — 4,015 words
- `rg -n "this chapter|present chapter|guardrail|safe historical|safer move|Honesty Over Output|prompt" src/content/docs/ai-history/ch-29-support-vector-machines.md` — no matches
- `../../.venv/bin/python scripts/test_pipeline.py` — 166 tests passed
- `npm run build` from primary checkout at `origin/codex/394-ch29-prose` — passed, 1,853 pages built

### CODE REVIEW

**Comment by krisztiankoos** [comment]
## Review: Ch29 Support Vector Machines prose

**Verdict: APPROVE**

**Reviewer:** claude-sonnet-4-6 (cross-family review of Codex-authored prose, per `docs/review-protocol.md`).

---

### 1. Source fidelity — verified

All load-bearing claims cross-checked against the anchored primaries. I downloaded `https://homepages.math.uic.edu/~lreyzin/papers/cortes95.pdf` and ran `pdftotext` to confirm the OCR specifics in particular:

- **USPS split & resolution.** Chapter: `7,300 training examples and 2,000 test examples, with 16-by-16 images`. Cortes/Vapnik 1995 §6.2: *"a US Postal Service database that contains 7,300 training patterns and 2,000 test patterns. The resolution of the database is 16 x 16 pixels."* Match.
- **NIST split & resolution.** Chapter: `60,000 training examples and 10,000 test examples, with 28-by-28 images`. Paper §6.2: *"60,000 training and 10,000 test patterns... resolution of these patterns is 28 x 28."* Match. (Worth noting `sources.md` Scene-Level table only anchored the splits, not the dimensions — the dimensions are in fact in C/V 1995, so the anchor table could be tightened, but the prose is correct.)
- **1.1% error attribution.** Chapter: `1.1 percent test error on the NIST task for a fourth-degree polynomial classifier`. Paper p.633: *"The combined performance of the ten classifiers on the test set is 1.1% error"* (one-versus-rest decomposition for 10-class digit recognition). Standard reading, fine.
- **LeNet attribution.** Chapter explicitly attributes the comparison to Cortes/Vapnik's report (`those comparison figures are Cortes and Vapnik's report, not an independent reanalysis`). Paper p.635 references LeNet1 and LeNet4 in the comparison table. Guardrail satisfied.
- **Aizerman/Braverman/Rozonoer + Mercer lineage.** Chapter attributes via `Cortes and Vapnik later connected this to a longer lineage of potential functions and Mercer's theorem`. Anchored at C/V 1995 p.283 in `sources.md`. Match.
- **Capacity-not-dimensionality claim.** Chapter: `complexity depended on the number of support vectors rather than directly on the dimensionality of the feature space`. Anchored at p.296. Match.
- **VC-theory machinery.** Chapter: `induced subsamples, the index of a system, and the growth function`. Anchored at Vapnik/Chervonenkis 1971 pp.265-266. Match. Crucially, the prose stays conceptual — *"The technical details belong in a probability text, not in this chapter"* — and avoids any theorem derivation. Guardrail satisfied.

### 2. Guardrails — all respected

- No "SVMs defeated neural networks" framing. Chapter explicitly: `Their importance was not that they defeated neural networks. LeNet remained real. Backpropagation remained real.`
- No "1990s winner" story. Chapter: `SVMs did not end the 1990s with a permanent victory. No method did.` and `neither SVMs nor neural networks "won" the 1990s in isolation` (paraphrased in the brief, executed in §"What SVMs Changed").
- VC theory stays conceptual — no theorem statements, no proof sketches, no derivations.
- No unsupported lab/hardware color. There is no Bell Labs anecdote, no AT&T workstation detail, no compute timing claim. (The chapter resists temptations the brief specifically warned about.)
- No modern scikit-learn detour. None present.

### 3. 4k-word justification

Brief's prose-capacity plan budgets **4,000-5,500 words**; draft is **4,121 words**, sitting at the floor. Section-by-section, paragraphs build argument rather than restate it. I don't find dead padding. The "Margin as a Contract" and "Kernel Move" sections are slightly under their planned budgets (~600 words each vs. 800-1,100 planned), but they're tight, not thin. The "What SVMs Changed" coda is the most repetitive section — it does restate points from §"Soft Margins and Real Data" and §"OCR as Proof Terrain" — but it functions as the explicit guardrail closing (`SVMs did not end the 1990s with a permanent victory`), so the repetition has a purpose.

### 4. Style fit

Frontmatter (`title`, `description`, `slug`, `sidebar.order`) and the single `> [!note] Pedagogical Insight:` callout match the existing Ch33/Ch10 pattern. No emojis. No `47`. Code blocks N/A. Astro/Starlight admonition syntax is the same as Ch33's. Hands off cleanly with the closing reference to *"empirical speech recognition, reinforcement learning, web-scale data, multicore and distributed compute, human annotation markets, and eventually the ImageNet-era return"* — which lines up with Ch32 (DARPA SUR), Ch36 (multicore wall), Ch37 (distributing the compute), Ch38 (human API), Ch41 (graphics hack), Ch43 (ImageNet). Good handoff geometry.

One tonal note: 4,121 words is roughly 4× the length of existing AI-history chapters in the published `src/content/docs/ai-history/` directory (Ch33 is 937 words, Ch10 is 941). The brief explicitly plans for this length, so it's intentional, but readers moving from Ch28 → Ch29 → Ch32 will feel the asymmetry until Ch24/Ch28 land at comparable depth. Not blocking.

---

### Nits (non-blocking)

**Nit 1: Single-classifier vs. ten-classifier framing on 1.1% error**
- FILE:LINE: `src/content/docs/ai-history/ch-29-support-vector-machines.md:298` (the line `Cortes and Vapnik report strong polynomial support-vector network performance and a 1.1 percent test error on the NIST task for a fourth-degree polynomial classifier.`)
- CURRENT CODE: `1.1 percent test error on the NIST task for a fourth-degree polynomial classifier`
- WHY: Cortes/Vapnik p.633 attributes the 1.1% figure to *"the combined performance of the ten classifiers"* — the 10 one-versus-rest binary SVMs assembled into a multiclass digit classifier. The chapter's framing is conventional and not wrong, but a strict reader could read "a fourth-degree polynomial classifier" as a single binary classifier.
- FIX (optional): `1.1 percent test error on the NIST task for a fourth-degree polynomial support-vector network (assembled from ten one-versus-rest binary classifiers, in the convention of the paper)`. Or leave as-is — within bounds of normal paraphrase.

**Nit 2: `sources.md` Scene-Level Claim Table doesn't anchor the 16×16/28×28 dimensions**
- FILE: `docs/research/ai-history/chapters/ch-29-support-vector-machines-svms/sources.md` (Scene table row "USPS/NIST OCR gave SVMs benchmark credibility next to LeNet…")
- WHY: The prose claim `with 16-by-16 images` / `with 28-by-28 images` is in C/V 1995 §6.2 (verified above), but the row's `Notes` column only enumerates `7,300/2,000` and `60,000/10,000` and `1.1%`. Adding `USPS 16×16; NIST 28×28` would close the source-trail gap so a future auditor doesn't have to redo the PDF check.
- FIX: append to the Notes cell: `; USPS 16×16; NIST 28×28 (C/V 1995 §6.2)`.

**Nit 3: Repetition of "discipline" / "disciplined"**
- The word appears ~22 times across the chapter and is a load-bearing term, but a couple of recurrences in "What SVMs Changed" could be varied (e.g., "rigor", "constraint", "controlled framing") if a copy pass is in scope.
- Not blocking; this is a thematic word and the repetition is intentional.

---

### Summary

Prose is faithful to the anchored sources, respects every guardrail in the brief, and earns its length without padding. Dataset numbers, image resolutions, error rates, and the LeNet-attribution language all check out against Cortes/Vapnik 1995 directly. Recommend merge; the two nits above are optional polish.

— claude-sonnet-4-6 (cross-family review, Claude reviewing Codex-authored prose)

**Comment by krisztiankoos** [comment]
## Gemini Prose Review (first pass)

**Verdict: NEEDS_CHANGES**

Gemini's first prose review found the factual guardrails safe but identified prompt/meta leakage and repetition:

- final `Honesty Over Output` note repeated process constraints instead of reader-facing content;
- prose used internal review language like "guardrail", "safe historical claim", and "safer move";
- some repetition around discipline/trust/generalization should be tightened.

Fixes applied in `723645ad`:
- removed the final process note;
- rewrote the kernel-lineage and OCR-comparison paragraphs as direct historical prose;
- removed "guardrail" meta-language from the conclusion;
- tightened the chapter from 4,121 to 4,019 words while keeping it above the 4k target;
- applied Claude's OCR nits: 1.1% NIST result now names the multiclass one-vs-rest setup, and `sources.md` records the 16x16 / 28x28 image dimensions.

A focused Gemini follow-up is requested after this comment.

**Comment by krisztiankoos** [comment]
Gemini Flash follow-up review trail (after commit 723645ad):

Verdict: NEEDS_CHANGES.

Findings:
- Remaining meta reference: "The technical details belong in a probability text, not in this chapter..."
- Remaining meta reference in the LeNet comparison paragraph: "in this chapter".
- Nit: use "setup" rather than "system" for the NIST multiclass wording.

Fixes applied in commit ea80680b:
- Removed both remaining chapter/process meta references.
- Reworded the LeNet comparison as direct historical interpretation.
- Changed the NIST wording to "setup".
- Rechecked for forbidden/meta phrases with rg; no matches.
- Final word count: 4,015.

**Comment by krisztiankoos** [comment]
Gemini final follow-up review on commit ea80680b:

Model note: gemini-3.1-pro-preview availability check timed out, so the bridge used the requested fallback, gemini-3-flash-preview.

Verdict: APPROVE.

Gemini verified:
- Previous process/meta-framing blockers are resolved.
- Remaining "this chapter" references are gone.
- LeNet comparison is rewritten as direct historical interpretation.
- NIST wording now uses "setup".
- Word count is approximately 4,000 words.
- Source anchors/status are consistent with project standards.

No remaining blockers.
